File: objc/Adapter/Adapter/MobileDataConnection.m
#import "MobileDataConnection.h"

@implementation MobileDataConnection

-(void)makeReady {
    NSLog(@"Making MobileData Ready...");
}

-(BOOL)isReady {
    return (arc4random_uniform(100) % 2);
}

-(void)sendPacket:(NSString*)packet {
    if(packet.length > 5) return;
    NSLog(@"Packet sent. [%@]", packet);
}

@end

File: swift/Adapter/Adapter/MobileDataConnection.swift
import Foundation

class MobileDataConnection: Connection {
    
    func makeReady() {
        print("Making MobileData Ready...")
    }
    
    func isReady() -> Bool {
        return arc4random_uniform(2) == 0
    }
    
    func sendPacket(_ data: String) {
        if data.count > 10 { return }
        print("Packet sent. [\(data)]")
    }
    
}
